--- silo-1.4.13/first/Makefile.orig	2006-06-01 19:24:53.000000000 +0200
+++ silo-1.4.13/first/Makefile	2006-11-09 11:09:11.000000000 +0100
@@ -56,7 +56,7 @@
 ultra.b: ultra
 	$(ELFTOAOUT) -o ultra.b ultra
 	echo -n 'SILO'$(IMGVERSION) | $(DD) of=ultra.b bs=8 conv=notrunc seek=3 count=1
-	echo -n -e '\340' | $(DD) of=ultra.b bs=1 count=1 seek=7 conv=notrunc
+	echo -n -e '\0340' | $(DD) of=ultra.b bs=1 count=1 seek=7 conv=notrunc
 	$(DD) if=/dev/zero of=ultra.b bs=4 count=1 seek=127
 
 generic.o: generic.S
@@ -70,7 +70,7 @@
 generic.b: generic
 	$(ELFTOAOUT) -o generic.b generic
 	echo -n 'SILO'$(IMGVERSION) | $(DD) of=generic.b bs=8 conv=notrunc seek=3 count=1
-	echo -n -e '\340' | $(DD) of=generic.b bs=1 count=1 seek=7 conv=notrunc
+	echo -n -e '\0340' | $(DD) of=generic.b bs=1 count=1 seek=7 conv=notrunc
 	$(DD) if=/dev/zero of=generic.b bs=4 count=1 seek=255
 
 fd.o: fd.S
@@ -84,7 +84,7 @@
 fd.b: fd
 	$(ELFTOAOUT) -o fd.b fd
 	echo -n 'SILO'$(IMGVERSION) | $(DD) of=fd.b bs=8 conv=notrunc seek=3 count=1
-	echo -n -e '\340' | $(DD) of=fd.b bs=1 count=1 seek=7 conv=notrunc
+	echo -n -e '\0340' | $(DD) of=fd.b bs=1 count=1 seek=7 conv=notrunc
 	$(DD) if=/dev/zero of=fd.b bs=4 count=1 seek=255
 
 ieee32.o: ieee32.S ieee32.h
